src/module.indexer/model/dftx/resign.masternode.ts

```
@@ -23,17 +27,47 @@ export class ResignMasternodeIndexer extends DfTxIndexer<ResignMasternode> {
           resignHeight: block.height,
           resignTx: txn.txid
         })
+
+        await this.indexStats(block, mn)
       }
     }
   }
 
-  async invalidate (_: RawBlock, txns: Array<DfTxTransaction<ResignMasternode>>): Promise<void> {
+  async indexStats (block: RawBlock, data: Masternode): Promise<void> {
+    const latest = await this.masternodeStatsMapper.getLatest()
+
+    await this.masternodeStatsMapper.put({
+      id: HexEncoder.encodeHeight(block.height),
+      stats: {
+        count: (latest?.stats?.count ?? 0) - 1,
```

Undefined scenario here is actually fatal error, do we have any notification service yet?

---

I was aware of this when I made it this way. It should be impossible to be undefined, as you cannot resign without creating first. It should be fatal IMO as something would be very wrong.

---

Ok I get what you're saying, it should actually throw if undefined here?

---

Yeah, I saw but the way it's designed now it has to do this. `jellyfish-indexer` will use more obvious and clear/automated invalidation techniques via undo,redo,prune keys.